### ISSUE TITLE
"Get-DesktopList" not recognized

### ISSUE THREAD

**Issue by InfinitGit** [MScholtes/PSVirtualDesktop#30]
Hey, just started getting this error while using the SyncVirtualDesktop add on. It was working fine before.
![image](https://github.com/MScholtes/PSVirtualDesktop/assets/124841095/2c4ccd63-96e8-4813-8d0d-cba347fcb061)
I believe it started after a recent Windows update... Windows info is below!

Edition	Windows 11 Home
Version	22H2
Installed on	‎6/‎9/‎2023
OS build	22621.2215
Experience	Windows Feature Experience Pack 1000.22662.1000.0

Let me know if I can provide any further info to help!

**Comment by MScholtes** [comment]
OK, I have made two attempts to find out what changed:

The first attempt (attached VirtualDesktopV3.ps1) assumes that the API is the same from new Windows 11 build 22621 upwards.

The second attempt (attached VirtualDesktopV4.ps1) assumes that the API from new build 22621 onwards is the same as the API from Windows Insider Beta before 8 March 2023 and that another API applies from Insider Dev onwards.

Please remove the additional extension txt from the files and try the following in the directory of the scripts:

```powershell
. .\VirtualDesktopV3.ps1
Get-DesktopList
```

exit Powershell now, start a new Powershell session and then

```powershell
. .\VirtualDesktopV4.ps1
Get-DesktopList
````

Can you post what's happening?

Greetings

Markus

[VirtualDesktopV4.ps1.txt](https://github.com/MScholtes/PSVirtualDesktop/files/12425989/VirtualDesktopV4.ps1.txt)
[VirtualDesktopV3.ps1.txt](https://github.com/MScholtes/PSVirtualDesktop/files/12425990/VirtualDesktopV3.ps1.txt)

**Comment by InfinitGit** [comment]
Hey man, first of all while I have you, AWESOME work on this whole app. I'm getting picky wanting it to work with virtual desktops, but overall the project is sick and I can see you're super responsive so keep it up!

Back to the task at hand, I tried both VirtualDesktopV3 and V4 and both seem to work... The scripts enable successfully and it doesn't throw the error I was getting before (before, I could recreate it by hitting "refresh wallpaper", but that works fine now on both). I'll use v4 for now and let you know if anything changes but it seems both at least run without errors; were we expecting only one to work?

**Comment by Difegue** [comment]
V3 works for me, not V4. 🤔 Same build as OP, just running Pro instead of Home

**Comment by InfinitGit** [comment]
Ahhh I lied. So I've tried both, and neither has an error message, but they do not work. Basically when I switch desktops the background flashes between dark and light and goes back and forth when I switch to and from.

**Comment by MScholtes** [comment]
Hello @InfinitGit,

I don't have Windows 11 available and won't have one for the foreseeable future. I am therefore dependent on (clear) advices. 
What does not work?

Switching? Everything? Why and when does the background flash? Are you using the script of Timothy Johnson for your tests? 
He is doing great work, but I cannot tell anything to his scripts.

Greetings

Markus

**Comment by MScholtes** [comment]
I have published the new version 1.5.4 of VirtualDesktop in the Powershell Gallery. However, I am not sure whether all versions and commandlets work.

I ask for feedback.

Greetings

Markus

**Comment by InfinitGit** [comment]
Ah, I apologize Markus! I didn't even realize your script and Timothy's were different things.

How am I to use your script? My use case is the WinDynamicDesktop app, so I had been placing the script files in the "scripts" folder of that app. Do I need to install yours elsewhere? Also do I need both your's and Timothy's or just one or the other? Currently, my scripts folder is as below, please advise me on how to properly test your scripts above or version 1.54
![image](https://github.com/MScholtes/PSVirtualDesktop/assets/124841095/fddf10a9-47af-4189-8db0-f2162a673170)

Regarding the 'flashing', it is to do with the dynamic desktop app. When I move to the next desktop, the background changes time, but not correctly, for example if it is supposed to be night, when I switch to it, it goes from day to night immediately, and then fades to day. This happens when switching back and forth from each desktop. I can try to capture a video of this tomorrow. After I switch back and forth many times eventually they are all synced and then I cannot recreate the problem until the morning/the next time they are supposed to switch. 
The first desktop change in the video below exhibits the issue, but after that they are all synced and thus act normal. I can try to capture more of the issue in the morning.

https://github.com/MScholtes/PSVirtualDesktop/assets/124841095/d0b5ccdc-82d0-4460-ac28-0296b3f29f42

Thanks for the troubleshooting!

**Comment by Difegue** [comment]
1.5.4 works for me -- Thanks! It still pops up an error message but virtual desktop functionality itself works fine on my end.
![image](https://github.com/MScholtes/PSVirtualDesktop/assets/8237712/fc434b87-de78-4460-bdda-9577daedb0e3)

@InfinitGit just use https://github.com/t1m0thyj/WDD-scripts/blob/master/experimental/SyncVirtualDesktops.ps1, that'll automatically use the latest published version of PSVirtualDesktop.

**Comment by InfinitGit** [comment]
> @InfinitGit just use https://github.com/t1m0thyj/WDD-scripts/blob/master/experimental/SyncVirtualDesktops.ps1, that'll automatically use the latest published version of PSVirtualDesktop.

Ooh gotcha, thank you! So if I just keep his script installed it will pull the PSVirtualDesktop on its own and keep pulling the newest version?

If that's the case, then @MScholtes I still get the error below so I guess there is still an issue? The error does list v1.54 so it seems to have used the latest version itself to no avail.
![image](https://github.com/MScholtes/PSVirtualDesktop/assets/124841095/b3bc9ce7-0429-40af-83d8-70e2cf6b3f26)

**Comment by MScholtes** [comment]
Hello @InfinitGit,

I'm not the author of script [SyncVirtualDesktops.ps1](https://github.com/t1m0thyj/WDD-scripts/blob/master/experimental/SyncVirtualDesktops.ps1). Please ask the author first if he can give more information.

Greetings

Markus

**Comment by MScholtes** [comment]
Hello,

I released a new bug fix version 1.5.5 today - tested only against Windows 11 22H2.2215. 
According to Microsoft's will, this may become obsolete again with the September update.

It corrects issues with creating, removing and naming desktops on Windows 11 22H2.2215 (reason for the error: _SwitchDesktopAndMoveForegroundView_ is missing in 2215).

I do not know, if this works on Insider versions build 22631 and up. It would be helpful if you try
```powershell
New-Desktop
```
if you own an Insider version and post the result and the build number here.


Greetings

Markus

**Comment by MScholtes** [comment]
No further activities required
